source/manual/architecture-shallow-dive.html.md

```diff
@@ -0,0 +1,204 @@
+
+If Fastly has the search results page cached, Fastly returns the results page to the user.
+
+If Fastly does not have the search results page cached, Fastly sends the request on to Origin. Origin is the infrastructure that GOV.UK maintains, hosted in Amazon Web Services (AWS).
+
+Within Origin, the request goes to an application called Router. The Finder Frontend application serves search results pages, so Router sends the request to Finder Frontend.
+
+Finder Frontend queries the Search API. Search API provides a [JSON response with the search results](https://www.gov.uk/api/search.json?q=covid&count=2). Finder Frontend then renders the response in a search results page to the user.
+
+## A publisher publishes content on GOV.UK
+
+GOV.UK has the following applications to publish content:
+
+- Mainstream Publisher
+- Whitehall
+- Content Publisher
+- Specialist Publisher
+- Travel Advice Publisher
+- Collections Publisher
+
+The following content states what happens when a user publishes content to live and to draft using Mainstream Publisher ("Publisher").
+
+### A user publishes content to live
+
+A user signs into their Signon account. The user creates or changes content in Publisher and saves that content in draft.
+
+Publisher makes a request to Publishing API to create or update the appropriate draft content.
+
+When the user is ready to publish the content, they press __Publish__ in the Publisher which sends a request to Publishing API.
+
+The Publishing API then updates the Content Store with the new or updated content.
+
+Content Store updates Router to say there is now content at a particular URL and says which frontend application will render that content.
+
+The Publishing API tells the Search API that the content is live.
+
+Search API updates its indexes so the content can be searched for by users.
+
+### A user publishes content to draft
+
+A user signs into their Signon account. The user creates or changes content in Publisher and saves that content in draft.
+
+Publisher makes a request to Publishing API to create or update the appropriate draft content.
+
+Before publishing content to live, a user can preview this draft content. To enable this, the following applications have both draft and live data and environments:
+
+- Router
+- Content Store
+- Government Frontend
+
+The Publishing API updates the draft Content Store when a user updates content, and then updates the live Content Store when the user then publishes the content.
+
+Access to the draft environment is restricted. The Authenticating Proxy sits in front of the draft Router, and checks whether a user should have access. The Proxy does this either through Signon credentials, or through a secret token if it is for fact checkers.
```

Comment:
It's difficult to introduce the draft stack without over-burdening the reader with too much complexity at once. The section above does a good job with that, but I don't think it makes it clear enough what lives on the draft stack and what lives on the live stack.

I've had a go at structuring the section the way that makes most sense to me, taking some content from [the draft/live section of the deep-dive](https://docs.publishing.service.gov.uk/manual/architecture-deep-dive.html#draft-and-live-stacks). Let me know what you think:

```suggestion
So far, we've been talking about the __live stack__, which is user-facing. Before publishing content to live, a publisher can preview their content in __the draft stack__, which is guarded by [authenticating-proxy](#authenticating-proxy) so that only authenticated users can view the content. It does this either through Signon credentials, or through a secret token if it is for fact checkers.

The publishing apps, Publishing API and message queue all run exclusively in this draft stack. However, the draft stack also has its own copy of Router, Content Store, and all of the frontend apps. This enables publishers to see what their draft content will look like when rendered by the respective frontend application.

The only app that communicates with both the draft and live stacks is Publishing API. When a draft is saved in Publisher, Publishing API saves the content to the draft content store. When a draft is published in Publisher, Publishing API saves the content to the live content store, which 'publishes' the content and makes it visible to users.
```

I think my proposal highlights the important role that Publishing API has in this. What I'm not so keen on is the sudden realisation that the publishing apps we've been talking about to date, run in a separate stack, and sit behind authenticating-proxy/Signon, which we've only just started talking about here. I wonder if the draft stack stuff should move further up in the document? 🤔

Reply:
👍